internal/component/kafka/consumer.go

```diff
@@ -35,28 +36,110 @@ type consumer struct {
+		if isBulkSubscribe {
+			if len(messages) < consumer.k.bulkSubscribeConfig.MaxBulkCount {
+				messages = append(messages, message)
+			} else {
+				break
+			}
+		} else {
+			if consumer.k.consumeRetryEnabled {
+				if err := retry.NotifyRecover(func() error {
+					return consumer.doCallback(session, message)
+				}, b, func(err error, d time.Duration) {
+					consumer.k.logger.Warnf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v. Retrying...", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				}, func() {
+					consumer.k.logger.Infof("Successfully processed Kafka message after it previously failed: %s/%d/%d [key=%s]", message.Topic, message.Partition, message.Offset, asBase64String(message.Key))
+				}); err != nil {
+					consumer.k.logger.Errorf("Too many failed attempts at processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				}
+			} else {
+				err := consumer.doCallback(session, message)
+				if err != nil {
+					consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				}
+			}
+		}
+	}
+	if isBulkSubscribe {
 		if consumer.k.consumeRetryEnabled {
 			if err := retry.NotifyRecover(func() error {
-				return consumer.doCallback(session, message)
+				return consumer.doBulkCallback(session, messages, claim.Topic())
 			}, b, func(err error, d time.Duration) {
-				consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v. Retrying...", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				consumer.k.logger.Warnf("Error processing Kafka bulk messages: %s. Error: %v. Retrying...", claim.Topic(), err)
 			}, func() {
-				consumer.k.logger.Infof("Successfully processed Kafka message after it previously failed: %s/%d/%d [key=%s]", message.Topic, message.Partition, message.Offset, asBase64String(message.Key))
+				consumer.k.logger.Infof("Successfully processed Kafka message after it previously failed: %s", claim.Topic())
 			}); err != nil {
-				consumer.k.logger.Errorf("Too many failed attempts at processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				consumer.k.logger.Errorf("Too many failed attempts at processing Kafka message: %s. Error: %v.", claim.Topic(), err)
 			}
 		} else {
-			err := consumer.doCallback(session, message)
+			err := consumer.doBulkCallback(session, messages, claim.Topic())
 			if err != nil {
-				consumer.k.logger.Errorf("Error processing Kafka message: %s/%d/%d [key=%s]. Error: %v.", message.Topic, message.Partition, message.Offset, asBase64String(message.Key), err)
+				consumer.k.logger.Errorf("Error processing Kafka message: %s. Error: %v.", claim.Topic(), err)
```

Comment:
Just a note : The whole set of messages are retried here ....

Reply:
Yes, this is on the same lines, as we had previously discussed that resiliency policies / retries will not be in effect in this release. Though resiliency and retry are different but their handling would be same in case of bulk pub/sub. 
So, in effect - this is a separate work TODO for upcoming release.